mkt/webapps/views.py

```diff
@@ -93,7 +93,10 @@ def create(self, request, *args, **kwargs):
             raise exceptions.PermissionDenied('You do not own that app.')
 
         # Create app, user and fetch the icon.
-        obj = Webapp.from_upload(upload, is_packaged=is_packaged)
+        try:
+            obj = Webapp.from_upload(upload, is_packaged=is_packaged)
+        except serializers.ValidationError as e:
```

Comment:
`from_upload` can return a `ValidationError`, if left uncatched it throws a 500 error like above, what we want is a 400.